### PR TITLE
T17 - String Permutations

### PR DESCRIPTION
**Pull request by karthick-1205**

### CODE REVIEW

File: MainRepository/Program.cs [hunk 1]
```diff
@@ -1,7 +1,20 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
+﻿//String permutation
+Console.Write ("Enter the string: ");
+string str = Console.ReadLine ();
+HashSet<string> result = GetPermutations (str);
```
Comment (priyarathna) [inline MainRepository/Program.cs:8]: @karthick-1205 : Why 'Hash' is used?

File: MainRepository/Program.cs [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
-        Console.WriteLine("Hello, World!");
-    }
+﻿//String permutation
+Console.Write ("Enter the string: ");
+string str = Console.ReadLine ();
+HashSet<string> result = GetPermutations (str);
+Console.WriteLine ($"Permutation of {str} are " + string.Join (" ", result));
+static HashSet<string> GetPermutations (string str) {
+   HashSet<string> permutations = new ();
+   if (str.Length == 0) {
```
Comment (priyarathna) [inline MainRepository/Program.cs:13]: If it's an empty string, then print the appropriate message.

**Review by priyarathna — CHANGES_REQUESTED** [review]

File: MainRepository/Program.cs (hunk 1 above)
Comment (karthick-1205) [reply MainRepository/Program.cs:8]: Remove duplicate string.

File: MainRepository/Program.cs (hunk 2 above)
Comment (priyarathna) [reply MainRepository/Program.cs:13]: @karthick-1205 : Not yet handled.

**Review by priyarathna — CHANGES_REQUESTED** [review]

**Review by priyarathna — APPROVED** [review]

**Pull request closed without merging**